rest-client-base/src/main/java/com/wultra/core/rest/client/base/DefaultRestClient.java

```diff
@@ -0,0 +1,700 @@
+        return get(path, null, responseType);
+    }
+
+    @Override
+    public <T> ResponseEntity<T> get(String path, MultiValueMap<String, String> headers, ParameterizedTypeReference<T> responseType) throws RestClientException {
+        ClientResponse response;
+        try {
+            response = buildUri(webClient.get(), path)
+                    .headers(h -> {
+                        if (headers != null) {
+                            h.addAll(headers);
+                        }
+                    })
+                    .exchange()
+                    .block();
+            validateResponse(response, responseType);
+            return response.toEntity(responseType).block();
+        } catch (RestClientException ex) {
+            // Rethrow validation errors
+            throw ex;
+        } catch (Exception ex) {
+            throw new RestClientException("HTTP GET request failed", ex);
+        }
+    }
+
+    @Override
+    public void getNonBlocking(String path, Consumer<ClientResponse> onSuccess, Consumer<Throwable> onError) throws RestClientException {
+        getNonBlocking(path, null, onSuccess, onError);
+    }
+
+    @Override
+    public void getNonBlocking(String path, MultiValueMap<String, String> headers, Consumer<ClientResponse> onSuccess, Consumer<Throwable> onError) throws RestClientException {
+        try {
+            buildUri(webClient.get(), path)
+                    .headers(h -> {
+                        if (headers != null) {
+                            h.addAll(headers);
+                        }
+                    })
+                    .accept(config.getAcceptType())
+                    .exchange()
+                    .subscribe(onSuccess, onError);
+        } catch (Exception ex) {
+            throw new RestClientException("HTTP GET request failed", ex);
+        }
+    }
+
+    @Override
+    public <T> ObjectResponse<T> getObject(String path, Class<T> responseType) throws RestClientException {
+        return getObject(path, null, responseType);
+    }
+
+    @Override
+    public <T> ObjectResponse<T> getObject(String path, MultiValueMap<String, String> headers, Class<T> responseType) throws RestClientException {
+        ParameterizedTypeReference<ObjectResponse<T>> typeReference = getTypeReference(responseType);
+        ResponseEntity<ObjectResponse<T>> responseEntity = get(path, headers, typeReference);
+        return responseEntity.getBody();
+    }
+
+    @Override
+    public <T> ResponseEntity<T> post(String path, Object request, ParameterizedTypeReference<T> responseType) throws RestClientException {
+        return post(path, request, null, responseType);
+    }
+
+    @Override
+    public <T> ResponseEntity<T> post(String path, Object request, MultiValueMap<String, String> headers, ParameterizedTypeReference<T> responseType) throws RestClientException {
+        ClientResponse response;
```

Comment:
Can we move the declaration to the assignment?

Reply:
Fixed.